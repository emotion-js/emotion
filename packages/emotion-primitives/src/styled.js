// @flow
import * as React from 'react'
import { View, Text, Image, StyleSheet } from 'react-primitives'
import {
  channel,
  contextTypes,
  setTheme,
  testAlwaysTrue,
  pickAssign
} from './utils'
import { getStyles } from './getStyles'
import { convertToRNStyles } from './convertToRNStyles'
import {
  testPickPropsOnPrimitiveComponent,
  testPickPropsOnOtherComponent
} from './test-props'

function isPrimitiveComponent(component: React.ElementType) {
  switch (component) {
    case View:
    case Text:
    case Image: {
      return true
    }
  }
  return false
}

// Evaluate the styles and convert them to React Native using styled component context
function evalStyles(context, Comp, styles) {
  // Convert styles with or without interpolations to React Native (StyleSheet.create)
  let otherStyles = convertToRNStyles.call(context, styles)

  return getStyles.call(context, otherStyles, context.props)
}

type State = {
  theme: Object
} | null

/**
 * a function that returns a styled component which render styles on multiple targets with same code
 */
export function createStyled(component: React.ElementType) {
  let isPrimitive = isPrimitiveComponent(component)
  let pick = isPrimitive
    ? testPickPropsOnPrimitiveComponent
    : testPickPropsOnOtherComponent
  return function createStyledComponent(...styles: *) {
    class Styled extends React.Component<*, State> {
      unsubscribe: number | void
      mergedProps: Object

      static withComponent = (newComponent: React.ElementType) =>
        createStyled(newComponent)(...styles)

      componentWillMount() {
        if (this.context[channel] !== undefined) {
          this.unsubscribe = this.context[channel].subscribe(
            setTheme.bind(this)
          )
        }
      }

      componentWillUnmount() {
        if (this.unsubscribe !== undefined) {
          this.context[channel].unsubscribe(this.unsubscribe)
        }
      }

      render() {
        const { props, state } = this

        // Similar to create-emotion-styled component implementation
        this.mergedProps = pickAssign(testAlwaysTrue, {}, props, {
          theme: (state !== null && state.theme) || props.theme || {}
        })

        const emotionStyles = evalStyles(this, Styled, styles)

        return React.createElement(
          component,
          pickAssign(pick, {}, props, {
            ref: props.innerRef,
            style: StyleSheet.flatten(emotionStyles)
          })
        )
      }
    }

    Styled.contextTypes = contextTypes

    Styled.displayName = `emotion(${getDisplayName(component)})`

    return Styled
  }
}

const getDisplayName = primitive =>
  typeof primitive === 'string'
    ? primitive
    : primitive.displayName || primitive.name || 'Styled'
