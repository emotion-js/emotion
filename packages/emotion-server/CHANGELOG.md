# emotion-server

## 10.0.27

### Patch Changes

- [`4c62ae9`](https://github.com/emotion-js/emotion/commit/4c62ae9447959d438928e1a26f76f1487983c968) [#1698](https://github.com/emotion-js/emotion/pull/1698) Thanks [@Andarist](https://github.com/Andarist)! - Add LICENSE file
- Updated dependencies [[`4c62ae9`](https://github.com/emotion-js/emotion/commit/4c62ae9447959d438928e1a26f76f1487983c968)]:
  - babel-plugin-emotion@10.0.27
  - create-emotion-server@10.0.27
  - emotion@10.0.27

## 10.0.17

### Patch Changes

- [16ff2330](https://github.com/emotion-js/emotion/commit/16ff233061e35fe71bfb1671da54ac12d6fc9eeb) [#1483](https://github.com/emotion-js/emotion/pull/1483) Thanks [@JakeGinnivan](https://github.com/JakeGinnivan)! - nth selector will no longer warn when using extract critical

## 10.0.14

### Patch Changes

- [c0eb604d](https://github.com/emotion-js/emotion/commit/c0eb604d) [#1419](https://github.com/emotion-js/emotion/pull/1419) Thanks [@mitchellhamilton](https://github.com/mitchellhamilton)! - Update build tool
