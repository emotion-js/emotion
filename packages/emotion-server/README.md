# emotion-server

> Extract and inline critical css with emotion.

`emotion-server` provides three APIs for doing server-side rendering with emotion to extract critical css, inline critical css in html to a string and inline critical css in html to a stream.

[`emotion-server`'s APIs are documented here.](https://emotion.sh/docs/ssr)

```bash
npm install --save emotion emotion-server
```
