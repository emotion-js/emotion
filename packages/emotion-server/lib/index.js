'use strict';

exports.__esModule = true;
exports.renderStylesToNodeStream = exports.renderStylesToString = exports.extractCritical = undefined;

var _createEmotionServer2 = require('create-emotion-server');

var _createEmotionServer3 = _interopRequireDefault(_createEmotionServer2);

var _emotion = require('emotion');

var emotion = _interopRequireWildcard(_emotion);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _createEmotionServer = (0, _createEmotionServer3.default)(emotion),
    extractCritical = _createEmotionServer.extractCritical,
    renderStylesToString = _createEmotionServer.renderStylesToString,
    renderStylesToNodeStream = _createEmotionServer.renderStylesToNodeStream;

exports.extractCritical = extractCritical;
exports.renderStylesToString = renderStylesToString;
exports.renderStylesToNodeStream = renderStylesToNodeStream;