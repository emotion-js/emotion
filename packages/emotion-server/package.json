{
  "name": "emotion-server",
  "version": "9.1.3",
  "description": "Extract and inline critical css with emotion for server side rendering.",
  "main": "lib/index.js",
  "types": "types/index.d.ts",
  "files": [
    "src",
    "lib",
    "types"
  ],
  "scripts": {
    "build": "npm-run-all clean babel",
    "babel": "babel src -d lib",
    "clean": "rimraf lib",
    "test:typescript": "dtslint types",
    "watch": "babel src -d lib --watch"
  },
  "dependencies": {
    "create-emotion-server": "^9.1.3"
  },
  "peerDependencies": {
    "emotion": "^9.1.3"
  },
  "devDependencies": {
    "@types/react-dom": "16.0.5",
    "babel-cli": "^6.24.1",
    "babel-plugin-emotion": "^9.1.2",
    "dtslint": "^0.3.0",
    "emotion": "^9.1.3",
    "npm-run-all": "^4.0.2",
    "react-emotion": "^9.1.3",
    "rimraf": "^2.6.1"
  },
  "author": "Kye Hohenberger",
  "homepage": "https://emotion.sh",
  "license": "MIT",
  "repository": "https://github.com/emotion-js/emotion/tree/master/packages/emotion-server",
  "keywords": [
    "styles",
    "emotion",
    "react",
    "css",
    "css-in-js",
    "ssr",
    "server-side-rendering"
  ],
  "bugs": {
    "url": "https://github.com/emotion-js/emotion/issues"
  }
}
