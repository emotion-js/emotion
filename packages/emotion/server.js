module.exports = require('emotion-server')
