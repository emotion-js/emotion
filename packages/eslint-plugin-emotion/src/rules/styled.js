module.exports = {
  meta: {
    fixable: 'code'
  },
  create(context) {
    return {
      ImportDeclaration(node) {
        if (
          node.source.value === 'react-emotion' ||
          node.source.value === 'preact-emotion'
        ) {
          let newImportPath =
            node.source.value === 'react-emotion'
              ? '@emotion/styled'
              : '@emotion/preact-styled'
          let quote = node.source.raw.charAt(0)
          context.report({
            node: node.source,
            message: `styled should be imported from ${newImportPath}`,
            fix:
              node.specifiers.length === 1 &&
              node.specifiers[0].type === 'ImportDefaultSpecifier'
                ? fixer => {
                    return fixer.replaceText(
                      node.source,
                      quote + newImportPath + quote
                    )
                  }
                : undefined
          })
        }
      }
    }
  }
}
