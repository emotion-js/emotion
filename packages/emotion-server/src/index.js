// @flow
import createEmotionServer from 'create-emotion-server'
import * as emotion from 'emotion'

export const {
  extractCritical,
  renderStylesToString,
  renderStylesToNodeStream
} = createEmotionServer(emotion)
