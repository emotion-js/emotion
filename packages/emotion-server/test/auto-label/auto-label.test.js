// @flow
/**
 * @jest-environment node
 */
import React from 'react'
import { renderToString } from 'react-dom/server'
import styled from 'react-emotion'
import { css, injectGlobal, keyframes, flush, hydrate } from 'emotion'
import { extractCritical } from 'emotion-server'
import { prettyifyCritical, getInjectedRules } from '../util'

const emotion = require('emotion')

const getComponents = () => {
  const color = 'red'

  injectGlobal`
    @font-face {
      font-family: 'Patrick Hand SC';
      font-style: normal;
      font-weight: 400;
      src: local('Patrick Hand SC'), local('PatrickHandSC-Regular'),
        url(https://fonts.gstatic.com/s/patrickhandsc/v4/OYFWCgfCR-7uHIovjUZXsZ71Uis0Qeb9Gqo8IZV7ckE.woff2)
          format('woff2');
      unicode-range: U+0100-024f, U+1-1eff, U+20a0-20ab, U+20ad-20cf,
        U+2c60-2c7f, U+A720-A7FF;
    }
  `

  const bounce = keyframes`
    from, 20%, 53%, 80%, to {
      animation-timing-function: cubic-bezier(0.215, 0.610, 0.355, 1.000);
      transform: translate3d(0,0,0);
    }

    40%, 43% {
      animation-timing-function: cubic-bezier(0.755, 0.050, 0.855, 0.060);
      transform: translate3d(0, -30px, 0);
    }

    70% {
      animation-timing-function: cubic-bezier(0.755, 0.050, 0.855, 0.060);
      transform: translate3d(0, -15px, 0);
    }

    90% {
      transform: translate3d(0,-4px,0);
    }
  `

  const hoverStyles = css`
    color: hotpink;
    &:hover {
      color: white;
      background-color: lightgray;
      border-color: aqua;
      box-shadow: -15px -15px 0 0 aqua, -30px -30px 0 0 cornflowerblue;
    }
  `

  const Main = styled.main`
    ${hoverStyles};
    display: flex;
  `

  const Image = styled.img`
    animation: ${bounce};
    border-radius: 50%;
    height: 50px;
    width: 50px;
    background-color: ${color};
  `

  // this will not be included since it's not used
  css`
    display: none;
  `

  // this will be included in both because it doesn't have the css- prefix

  injectGlobal`
    .no-prefix {
      display: flex;
      justify-content: center;
    }
  `

  const Page1 = () => (
    <Main>
      <Image size={30} />
      <Image size={100} />
      <Image />
    </Main>
  )

  const Page2 = () => (
    <Main>
      <div>Hello</div>
    </Main>
  )
  return { Page1, Page2 }
}

describe('extractCritical', () => {
  test('returns static css', () => {
    const { Page1, Page2 } = getComponents()
    expect(
      prettyifyCritical(extractCritical(renderToString(<Page1 />)))
    ).toMatchSnapshot()
    expect(
      prettyifyCritical(extractCritical(renderToString(<Page2 />)))
    ).toMatchSnapshot()
  })
})
describe('hydration', () => {
  test('only rules that are not in the critical css are inserted', () => {
    const { Page1 } = getComponents()
    const { html, ids, css } = extractCritical(renderToString(<Page1 />))
    expect(prettyifyCritical({ html, css, ids })).toMatchSnapshot()
    flush()

    hydrate(ids)

    const { Page1: NewPage1 } = getComponents()
    renderToString(<NewPage1 />)
    expect(getInjectedRules(emotion)).toMatchSnapshot()
  })
})
