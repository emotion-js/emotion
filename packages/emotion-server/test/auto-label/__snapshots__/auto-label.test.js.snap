// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractCritical returns static css 1`] = `
Object {
  "css": "@font-face {
  font-family: 'Patrick Hand SC';
  font-style: normal;
  font-weight: 400;
  src: local('Patrick Hand SC'),local('PatrickHandSC-Regular'),url(https://fonts.gstatic.com/s/patrickhandsc/v4/OYFWCgfCR-7uHIovjUZXsZ71Uis0Qeb9Gqo8IZV7ckE.woff2) format('woff2');
  unicode-range: U+0100-024f,U+1-1eff,U+20a0-20ab,U+20ad-20cf,U+2c60-2c7f,U+A720-A7FF;
}

@-webkit-keyframes animation-1938e3k-bounce {
  from, 20%, 53%, 80%, to {
    -webkit-animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  40%, 43% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-30px,0);
    -ms-transform: translate3d(0,-30px,0);
    transform: translate3d(0,-30px,0);
  }

  70% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-15px,0);
    -ms-transform: translate3d(0,-15px,0);
    transform: translate3d(0,-15px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-4px,0);
    -ms-transform: translate3d(0,-4px,0);
    transform: translate3d(0,-4px,0);
  }
}

@keyframes animation-1938e3k-bounce {
  from, 20%, 53%, 80%, to {
    -webkit-animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  40%, 43% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-30px,0);
    -ms-transform: translate3d(0,-30px,0);
    transform: translate3d(0,-30px,0);
  }

  70% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-15px,0);
    -ms-transform: translate3d(0,-15px,0);
    transform: translate3d(0,-15px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-4px,0);
    -ms-transform: translate3d(0,-4px,0);
    transform: translate3d(0,-4px,0);
  }
}

.no-prefix {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.css-ggqssp-Main {
  color: hotpink;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.css-ggqssp-Main:hover {
  color: white;
  background-color: lightgray;
  border-color: aqua;
  box-shadow: -15px -15px 0 0 aqua,-30px -30px 0 0 cornflowerblue;
}

.css-xucs6l-Image {
  -webkit-animation: animation-1938e3k-bounce;
  animation: animation-1938e3k-bounce;
  border-radius: 50%;
  height: 50px;
  width: 50px;
  background-color: red;
}",
  "html": 
<main class="css-ggqssp-Main ev1dpeo0"
      data-reactroot
>
  <img size="30"
       class="css-xucs6l-Image ev1dpeo1"
  >
  <img size="100"
       class="css-xucs6l-Image ev1dpeo1"
  >
  <img class="css-xucs6l-Image ev1dpeo1">
</main>
,
  "ids": Array [
    "1s6lroo-getComponents",
    "1938e3k-bounce",
    "11lstga-getComponents",
    "ggqssp-Main",
    "xucs6l-Image",
  ],
}
`;

exports[`extractCritical returns static css 2`] = `
Object {
  "css": "@font-face {
  font-family: 'Patrick Hand SC';
  font-style: normal;
  font-weight: 400;
  src: local('Patrick Hand SC'),local('PatrickHandSC-Regular'),url(https://fonts.gstatic.com/s/patrickhandsc/v4/OYFWCgfCR-7uHIovjUZXsZ71Uis0Qeb9Gqo8IZV7ckE.woff2) format('woff2');
  unicode-range: U+0100-024f,U+1-1eff,U+20a0-20ab,U+20ad-20cf,U+2c60-2c7f,U+A720-A7FF;
}

@-webkit-keyframes animation-1938e3k-bounce {
  from, 20%, 53%, 80%, to {
    -webkit-animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  40%, 43% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-30px,0);
    -ms-transform: translate3d(0,-30px,0);
    transform: translate3d(0,-30px,0);
  }

  70% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-15px,0);
    -ms-transform: translate3d(0,-15px,0);
    transform: translate3d(0,-15px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-4px,0);
    -ms-transform: translate3d(0,-4px,0);
    transform: translate3d(0,-4px,0);
  }
}

@keyframes animation-1938e3k-bounce {
  from, 20%, 53%, 80%, to {
    -webkit-animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  40%, 43% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-30px,0);
    -ms-transform: translate3d(0,-30px,0);
    transform: translate3d(0,-30px,0);
  }

  70% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-15px,0);
    -ms-transform: translate3d(0,-15px,0);
    transform: translate3d(0,-15px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-4px,0);
    -ms-transform: translate3d(0,-4px,0);
    transform: translate3d(0,-4px,0);
  }
}

.no-prefix {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.css-ggqssp-Main {
  color: hotpink;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.css-ggqssp-Main:hover {
  color: white;
  background-color: lightgray;
  border-color: aqua;
  box-shadow: -15px -15px 0 0 aqua,-30px -30px 0 0 cornflowerblue;
}",
  "html": 
<main class="css-ggqssp-Main ev1dpeo0"
      data-reactroot
>
  <div>
    Hello
  </div>
</main>
,
  "ids": Array [
    "1s6lroo-getComponents",
    "1938e3k-bounce",
    "11lstga-getComponents",
    "ggqssp-Main",
  ],
}
`;

exports[`hydration only rules that are not in the critical css are inserted 1`] = `
Object {
  "css": "@font-face {
  font-family: 'Patrick Hand SC';
  font-style: normal;
  font-weight: 400;
  src: local('Patrick Hand SC'),local('PatrickHandSC-Regular'),url(https://fonts.gstatic.com/s/patrickhandsc/v4/OYFWCgfCR-7uHIovjUZXsZ71Uis0Qeb9Gqo8IZV7ckE.woff2) format('woff2');
  unicode-range: U+0100-024f,U+1-1eff,U+20a0-20ab,U+20ad-20cf,U+2c60-2c7f,U+A720-A7FF;
}

@-webkit-keyframes animation-1938e3k-bounce {
  from, 20%, 53%, 80%, to {
    -webkit-animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  40%, 43% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-30px,0);
    -ms-transform: translate3d(0,-30px,0);
    transform: translate3d(0,-30px,0);
  }

  70% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-15px,0);
    -ms-transform: translate3d(0,-15px,0);
    transform: translate3d(0,-15px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-4px,0);
    -ms-transform: translate3d(0,-4px,0);
    transform: translate3d(0,-4px,0);
  }
}

@keyframes animation-1938e3k-bounce {
  from, 20%, 53%, 80%, to {
    -webkit-animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    animation-timing-function: cubic-bezier(0.215,0.610,0.355,1.000);
    -webkit-transform: translate3d(0,0,0);
    -ms-transform: translate3d(0,0,0);
    transform: translate3d(0,0,0);
  }

  40%, 43% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-30px,0);
    -ms-transform: translate3d(0,-30px,0);
    transform: translate3d(0,-30px,0);
  }

  70% {
    -webkit-animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    animation-timing-function: cubic-bezier(0.755,0.050,0.855,0.060);
    -webkit-transform: translate3d(0,-15px,0);
    -ms-transform: translate3d(0,-15px,0);
    transform: translate3d(0,-15px,0);
  }

  90% {
    -webkit-transform: translate3d(0,-4px,0);
    -ms-transform: translate3d(0,-4px,0);
    transform: translate3d(0,-4px,0);
  }
}

.no-prefix {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.css-ggqssp-Main {
  color: hotpink;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.css-ggqssp-Main:hover {
  color: white;
  background-color: lightgray;
  border-color: aqua;
  box-shadow: -15px -15px 0 0 aqua,-30px -30px 0 0 cornflowerblue;
}

.css-xucs6l-Image {
  -webkit-animation: animation-1938e3k-bounce;
  animation: animation-1938e3k-bounce;
  border-radius: 50%;
  height: 50px;
  width: 50px;
  background-color: red;
}",
  "html": 
<main class="css-ggqssp-Main ev1dpeo0"
      data-reactroot
>
  <img size="30"
       class="css-xucs6l-Image ev1dpeo1"
  >
  <img size="100"
       class="css-xucs6l-Image ev1dpeo1"
  >
  <img class="css-xucs6l-Image ev1dpeo1">
</main>
,
  "ids": Array [
    "1s6lroo-getComponents",
    "1938e3k-bounce",
    "11lstga-getComponents",
    "ggqssp-Main",
    "xucs6l-Image",
  ],
}
`;

exports[`hydration only rules that are not in the critical css are inserted 2`] = `
".css-84uqt3-hoverStyles {
  color: hotpink;
}

.css-84uqt3-hoverStyles:hover {
  color: white;
  background-color: lightgray;
  border-color: aqua;
  box-shadow: -15px -15px 0 0 aqua,-30px -30px 0 0 cornflowerblue;
}

.css-18s9rqr-getComponents {
  display: none;
}"
`;
